docs/debugging.md

```diff
@@ -0,0 +1,143 @@
+pod. Find out what node the pod is scheduled on by passing `-o wide` and
+looking for `NODE`:
+
+```cli
+$ kubectl get pods -o wide
+NAME    READY   STATUS              RESTARTS   AGE   IP       NODE                                       NOMINATED NODE   READINESS GATES
+mypod   0/1     ContainerCreating   0          7s    <none>   gke-cluster-a-default-pool-3e0d8eb7-d2qk   <none>           <none>
+```
+
+Then look for the `csi-secrets-store` and `csi-secrets-store-provider-gcp` pod
+names on the same node:
+
+```cli
+$ kubectl get pods --all-namespaces --field-selector spec.nodeName=gke-cluster-a-default-pool-3e0d8eb7-d2qk
+NAMESPACE     NAME                                                        READY   STATUS              RESTARTS   AGE
+default       mypod                                                       0/1     ContainerCreating   0          2m11s
+kube-system   csi-secrets-store-provider-gcp-n9c5f                        1/1     Running             0          23h
+kube-system   csi-secrets-store-t9www                                     3/3     Running             0          4d3h
+...
+```
+
+## Logs
+
+The plugin uses the `klog` package to write logs to stderr. These are written as
+`json` so that they can be properly parsed by Cloud Logging.
+
+The following Cloud Logging filter will find logs from all
+`csi-secrets-store-provider-gcp` pods:
+
+```text
+resource.labels.project_id="<project>"
+resource.labels.location="<cluster location>"
+resource.labels.cluster_name="<cluster name>"
+resource.labels.namespace_name="kube-system"
+labels.k8s-pod/app="csi-secrets-store-provider-gcp"
+```
+
+If you are having trouble with a single pod, you can expand the filter to cover
+both the driver, plugin, and the pod pod:
+
+```text
+labels.k8s-pod/app=("csi-secrets-store-provider-gcp" OR "cloud-secrets-store")
+jsonPayload.pod="default/mypod"
+```
+
+`kubectl` will also show you logs for a specific driver/plugin pod:
+
+```cli
+kubectl logs csi-secrets-store-provider-gcp-4jljs --namespace=kube-system
+```
+
+To increase verbosity of logs edit `deploy/provider-gcp-plugin.yaml` to set
+`-v=5`:
+
+```yaml
+containers:
+- name: provider
+    ...
+    args:
+    - "-v=5"
+```
+
+## Metrics
+
+View container status metrics on GKE by navigating to the 
+`csi-secrets-store-provider-gcp` DeamonSet details page (`Workloads` then remove
+the `Is system object` filter).
+
+If the CPU or memory usage graphs appears to exceed limits then you may need to
+tune `deploy/provider-gcp-plugin.yaml` and request more resources for the pods:
+
+```yaml
+...
+resources:
+requests:
+    cpu: 50m
+    memory: 100Mi
+limits:
+    cpu: 50m
+    memory: 100Mi
+...
+```
+
+Prometheus metrics are served from port 8080, but this port is not exposed
+outside the pod by default. Use `kubectl port-forward` to access the
+metrics over localhost:
+
+```cli
+kubectl port-forward csi-secrets-store-provider-gcp-vmqct --namespace=kube-system 8080:8080
```

Comment:
Is this 8095 or 8080? I see it overridden in the yaml tmpl

Reply:
oo shoot i'll make this consistent